packages/core-mobile/app/new/common/hooks/usePinOrBiometryLogin.ts

```diff
@@ -156,12 +156,27 @@ export function usePinOrBiometryLogin({
         const accessType = BiometricsSDK.getAccessType()
 
         if (accessType === 'BIO') {
+          // Check if migration is needed first
+          const migrator = new KeychainMigrator(activeWalletId)
+          const result = await migrator.migrateIfNeeded('BIO')
+          if (
+            result.success &&
+            result.value === MigrationStatus.RunBiometricMigration
+          ) {
+            //already prompted user for bio, assume verified
+            setVerified(true)
+            resetRateLimiter()
+            return new NothingToLoad()
+          }
+          if (result.success) {
+            throw new Error(
+              'Invalid state: migration status is not RunBiometricMigration'
+            )
+          }
```

Comment:
do we actually want to throw an error for any status here? what if the status is no migration needed?